pennylane/tape/__init__.py

```diff
@@ -40,6 +41,128 @@
 _mock_stack = []
 
 
+class TapeOperationRecorder(QuantumTape):

```

Comment:
The `OperationRecorder` and `@template` decorator were never ported to work with the new queuing refactor.

As a result, in tape mode, they no longer work. They do not error, but instead just return empty lists, as they are not accessing the new `AnnotatedQueue` and `QueuingContext`. The versions below are written to work in tape mode, are dynamically patch the existing versions when `enable_tape` is called.